src/pages/index.js

```diff
@@ -1,67 +1,214 @@
 });
+
 // -------------------------------------------------------------------------------------
-//                                 Popup Instances
+// Inital UserInfo Population
 // -------------------------------------------------------------------------------------
-const profileEditPopup = new PopupWithForm("#profile-edit-modal", {
-  handleFormSubmit: (formData) => {
+api
+  .getProfileData()
+  .then((data) => {
     userInfo.setUserInfo({
-      name: formData["name"],
-      description: formData["description"],
+      name: data.name,
+      about: data.about,
+      avatar: data.avatar,
     });
-    profileEditPopup.close();
+  })
+  .catch((error) => {
+    console.error("Failed to fetch profile data:", error);
+  });
+
+// -------------------------------------------------------------------------------------
+// Editing UserInfo Popup
+// -------------------------------------------------------------------------------------
+const profileEditPopup = new PopupWithForm("#profile-edit-modal", {
+  handleFormSubmit: (formData) => {
+    profileEditPopup.toggleUploadIndicator(true);
+
+    api
+      .editProfileData(formData)
+      .then((data) => {
+        userInfo.setUserInfo(data);
+      })
+      .catch((error) => {
+        console.log("Error updating Profile:", error);
+      })
+      .finally(() => {
+        profileEditPopup.toggleUploadIndicator(false);
+        profileEditPopup.close();
+      });
   },
 });
+
 profileEditPopup.setEventListeners();
 
+// -------------------------------------------------------------------------------------
+// Editing UserAvatar Popup
+// -------------------------------------------------------------------------------------
+const avatarEditPopup = new PopupWithForm("#avatar-edit-modal", {
+  handleFormSubmit: (formData) => {
+    const avatarURL = formData["Image URL"];
+    avatarEditPopup.toggleUploadIndicator(true);
+
+    api
+      .editUserAvatar({ avatar: avatarURL })
+      .then((data) => {
+        userInfo.setUserInfo({
+          avatar: data.avatar,
+        });
+      })
+      .catch((error) => {
+        console.log("Error updating Avatar:", error);
+      })
+      .finally(() => {
+        avatarEditPopup.toggleUploadIndicator(false);
+        avatarEditPopup.close();
+      });
+  },
+});
+
+avatarEditPopup.setEventListeners();
+
+// -------------------------------------------------------------------------------------
+// Card Section Instance
+// -------------------------------------------------------------------------------------
+const cardSection = new Section(
+  {
+    items: [],
+    renderer: (cardData) => {
+      const card = new Card(
+        cardData,
+        cardTemplate,
+        (cardData) => imagePopup.open(cardData),
+        handleTrashClick,
+        api
+      ).getView();
+      if (cardData.isLiked) {
+        card
+          .querySelector(".card__like-button")
+          .classList.add("card__like-button_active");
```

Comment:
Now only class `Card` knows every **card selector**, so you need to encapsulate the likes logics inside its methods and call them in `index.js`. It’s a violation of **Single responsibility** principle


You can't search the like button here. It's been found inside `Card`, so you need to do it inside `Card`